torch/csrc/jit/init.cpp

```diff
@@ -339,6 +340,20 @@ void initJITBindings(PyObject* module) {
             auto stack = toTraceableStack(args);
             checkAliasAnnotation(g, std::move(stack), unqualified_op_name);
           })
+      .def(
+          "_jit_set_cuda_fusion_group_executor",
+          [](bool pass_flag) {
+            bool oldState = getCudaFusionGroupExecutorMode();
+            getCudaFusionGroupExecutorMode() = pass_flag;
+            return oldState;
+          })
+      .def(
+          "_jit_set_cuda_fusion_group_pass",
```

Comment:
Some knob (`_jit_override_can_fuse_on_gpu `) was just landed on master to disable CUDA fuser - do we need another one? Can we just reuse it?

Reply:
These are different knobs. The legacy fuser is supposed to be handling CPU as well as CUDA fusion, hence the knob to disable CUDA fuser.

We are adding an extra optimization pass that is dedicated for fusion on CUDA. Reusing the knob would make the control correlated, which is not what we want.

Reply:
Knobs for CPU and GPU are separate: `_jit_override_can_fuse_on_gpu` doesn't affect fusion on CPU. The CPU one (_jit_override_can_fuse_on_cpu) has been there for quite some time and defaults to `false`, the GPU one has just been added. So, I'm not sure we need yet another one :)

Reply:
I added two knobs here. `_jit_set_cuda_fusion_group_executor` is for debugging purpose, I can remove it if we hate extra knob so much.

`_jit_set_cuda_fusion_group_pass` controls whether or not to run the new optimization pass. I don't see an easy way to avoid it. Even in the proposed PR, we need a switch/knob to turn on/off the new pass: https://github.com/pytorch/pytorch/pull/33464/files#diff-eb506c4008838cc3262b15d62f8f4a62R326

Reply:
> `_jit_set_cuda_fusion_group_pass` controls whether or not to run the new optimization pass. I don't see an easy way to avoid it.

Yeah, this one is needed. But the first one I think is redundant. For debugging you can
1)  disable the current fuser with `_jit_override_can_fuse_on_gpu`
2) register your pass as a custom pass if `_jit_set_cuda_fusion_group_pass` is set

Reply:
updated as custom passes.
(FYI, the debugging I'm referring to is switching the codegen for CudaFusionGroup, but that's not a big deal and can easily done in code. Thanks for the suggestion).